docs/performance/inserts/bulk.txt

```diff
@@ -0,0 +1,147 @@
+.. _bulk_inserts:
+
+============
+Bulk Inserts
+============
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Single Inserts
+==============
+
+`Single inserts`_ use the ``INSERT`` statement, and look like this:
+
+.. code-block:: psql
+
+    cr> INSERT INTO my_table (column_a) VALUES ("value 1");
+    INSERT OK, 1 row affected (... sec)
+
+Single inserts are typically very fast with CrateDB. A small cluster can
+easily handle several thousand inserts per second.
+
+However, it's important to note that:
+
+ 1. Every insert is first applied to the primary shard
+
+ 2. After the primary shard has been updated, the insert is then individually
+    communicated in parallel to every configured replica shard
+
+ 3. CrateDB will not return a response until all replica shards have been
+    updated
+
+The overhead for each insert (query parsing, planning, and execution, which
+includes network connection setups, connection teardowns, and connection
+metadata) starts to add up for very heavy insert workloads.
+
+In addition, lots of internal traffic will congest your network, which will
+slow down any network-based cluster operations (i.e. inserts, distributed
+reads, replication, and cluster management).
+
+Fortunately, there is a solution: bulk inserts.
+
+Switching to Bulk Inserts
+=========================
+
+If you can batch up your inserts, switching to *bulk inserts* will dramatically
+improve both ingestion performance, and overall cluster performance.
+
+Bulk inserts make use of the CrateDB HTTP endpoint `bulk operations`_ feature
+to perform many inserts in a single operation.
+
+Bulk inserts still generate internal network traffic, and CrateDB still waits
+until all replicas have been updated before returning a response. But there is
+less internal network traffic. And in addition, the bulk query only needs to be
```

Comment:
maybe it's worth mentioning that the disk flush only happens once as well (if set to request, which is the default, see https://crate.io/docs/crate/reference/en/latest/sql/reference/create_table.html?highlight=flush#translog-durability)